ws-server/src/server/module.rs

```diff
@@ -35,16 +38,17 @@ impl RpcModule {
 	pub fn register_method<F, R>(&mut self, method_name: &'static str, callback: F) -> Result<(), Error>
 	where
 		R: Serialize,
-		F: RpcMethod<R>,
+		F: RpcMethod<R, InvalidParams>,
```

Comment:
Should we do something like `dyn Error` here instead? Or implement our own error trait that gives us the JSON-RPC error code along with the message?

Reply:
> Or implement our own error trait that gives us the JSON-RPC error code along with the message?

That sounds neat and I like but it might be annoying users to explicitly implement that trait for the error types that they "plug-in" in `Error::ContextFailed(user err)`? However, it would be a way for the users to explicitly add there own error codes for this

Reply:
Let's create an issue for this and address it another PR